sdk/pinocchio/src/cpi.rs

```diff
@@ -182,6 +182,116 @@ pub fn slice_invoke_signed(
     Ok(())
 }
 
+/// Invoke a cross-program instruction with signatures.
+///
+/// # Important
+///
+/// The accounts on the `account_infos` slice must be in the same order as the
+/// `accounts` field of the `instruction`.
+///
+/// This function does not check that [`AccountInfo`]s are properly borrowable.
+/// Those checks consume CPU cycles that this function avoids.
+///
+/// # Safety
+///
+/// If any of the writable accounts passed to the callee contain data that is
+/// borrowed within the calling program, and that data is written to by the
+/// callee, then Rust's aliasing rules will be violated and cause undefined
+/// behavior.
+pub unsafe fn invoke_signed_access_unchecked<const ACCOUNTS: usize>(
+    instruction: &Instruction,
```

Comment:
It could be worth creating a `FixedInstruction` variant that takes in const generic for the account metas length. constraining the number of accounts to account metas at compile time.